src/nexuscli/cli/subcommand_repository.py

```diff
@@ -24,6 +23,9 @@
          [--blob=<store_name>] [--strict-content] [--cleanup=<c_policy>]
          [--write=<w_policy>]
          [--depth=<repo_depth>]
+  nexus3 repository (delete|del) <repo_name> [--force]
+  nexus3 repository del_assets_regex <repo_name> <assets_regex> [--force]
```

Comment:
Wouldn't these `del_assets*` subcommands be better under the existing root command (`delete|del`)?

```
  nexus3 (delete|del) <repository_path>
```

You could add flags for the user to specify whether the `<repository_path>` contains a regex or a wildcard (glob?)

I'll have a better look at the problem this weekend so I have a better informed suggestion.

Reply:
Regarding the 2 points:
1) subcommand vs root command: honestly as user of nexus3-cli I found very confusing having both: indeed it was quite confusing to have these subcommands documented just at the end of the "--help" section and it took me a while to discover how many possibilities nexus3-cli provides beside the root commands.
So IMHO I would completely remove root commands and leave only subcommands: all the list/upload/download/delete root commands IMHO should be under an "assets" subcommand.
Then in this scenario these del_assets_* commands added by this PR could be merged with the "delete" root command perhaps.

2) regarding having a single command instead of del_assets_regex/del_assets_wildcard: I think you're right, having a flag would make it possible to unify them.
Should the default be to interpret the parameter appearing after the <repo_name> as a wildcard or as a regex? Since wildcards are easier to use I would opt for the former.

Reply:
1. All the commands used to be at the root level but I found that confusing as well. The help page was very long and most users want to upload/download/delete. So I decided to pack them in sub commands as this is already an established pattern in many cli utilities (git, for example).

As changing the CLI would be a breaking change, I’d favour a bit more research/discussion before acting on it. Perhaps build the docopt in a separate branch so we can see how it looks.

2. I agree wildcard makes more sense. There’s an existing behaviour where it deletes everything under a matching name if the name ends with a / (eg del directory/) - if possible, I’d like to keep this behaviour.

Thanks again for your contribution.

Reply:
1. Ok sure - I understand that grouping root commands all together would be a breaking change... better not to mix with this MR. So, for now, should I keep the del_assets command under the "repository" subcommand? Or do you prefer to have a new subcommand or a root-level command for that?

2. I have to say that after some more attempts to use the new "del_assets_wildcard" command I discovered that the OrientDB used by Nexus3 is handling wildcards in a weird (at least to me) way: i.e. the wildcard symbol % matches only string prefixes or postfixes. E.g. if I have an asset named "folder1/myasset-1.2.3.rpm" and I provide as wildcard "folder1/%1.2.3%" the Groovy script will report 0 matches. If I provide "%1.2.3%", the asset will be matched.
All this to say: maybe allowing the user to use wildcards instead of regex is not that useful finally...

Reply:
1. Yes please. I'd prefer to keep it consistent with the subcommand pattern. I'll take your feedback onboard and think of a way to better expose the subcommands to first-time users.

2. I think that's still useful - if we can clearly communicate how the wildcard handling works (or link to a place that does it). One thing that I didn't consider is what happens when a file you want to delete has a wildcard character in it. I guess the user would need to escape it by default, unless we only enable these options through flags; e.g.:

```
  nexus3 (delete|del) <repository_path> [--regex|--wildcard] 

Options:
  --regex          Enable regular expression parsing on <repository_path>
  --wildcard       Enable wildcard parsing on <repository_path>
```

Reply:
Hi @thiagofigueiro ,
just to be sure I understand: from reading the example docs you put in point #2 I assume that the answer w.r.t. point #1 is "having these del_assets_wildcard and del_assets_regex" merged with the root-level "del" command right?

Reply:
Hi @thiagofigueiro ,
I was going to implement the change but I realized now that regarding point 1
  1. if you want to have the "del_assets_wildcard/regex" commands merged with the "delete|del" command there's a small detail to fix:  former cmds take the repository name and, separated, the repository path. The latter cmd is taking just the repository path. So how are user supposed to delete assets by regex?
Would that be something like:
`
   nexus3 del --regex myrepoName/^[a-z]*myregex.*
`
?
Or can we modify the "delete|del" command to separate the reponame from the asset name:
`
   nexus3 del --regex myrepoName  ^[a-z]*myregex.*
`
?

Reply:
`--regex` and `--wildcard` would be toggle flags to the `del` command.

There's [existing code](https://github.com/thiagofigueiro/nexus3-cli/blob/master/src/nexuscli/nexus_client.py#L270) that can receive this
```
nexus3 del --regex myrepoName/^[a-z]*myregex.*
```

and split it using something [like this](https://github.com/thiagofigueiro/nexus3-cli/blob/master/src/nexuscli/nexus_client.py#L363).

It's a part of the code that is older and somewhat messy but it does the job.  Assuming this is the docopt line:
```
  nexus3 (delete|del) <repository_path>
```

Then you could do something like this:

```
if options.get('--regex') or options.get('--wildcard'): 
    repository, regex_or_wildcard = self._pop_repository(options.get('<repository_path>'))
    # delete wildcard or regex code
else:
    # existing delete code
```

Reply:
Hi @thiagofigueiro ,
I'm trying to do the change but I admit I'm facing several issues as I don't know the code so well.
For example: the "nexus3 (delete|del) <repository_path>" command is not using the "api/repository/*" code, while my PR has added all the code to perform delete operations in the API first. 
Should I change the "delete|del" command to go through the API?
In practice such existing "delete|del" command can be implemented as a special case of the wildcard code where simply no wildcards are provided...